scte35/segmentationdescriptor.go

```diff
@@ -0,0 +1,258 @@
+		0x36: {0x30: segCloseDiffPTS, 0x32: segCloseDiffPTS, 0x36: segCloseNotNested},
+		0x37: {0x30: segCloseNormal, 0x32: segCloseNormal, 0x36: segCloseEventIDNotNested},
+		0x40: {0x10: segCloseNormal, 0x14: segCloseNormal, 0x17: segCloseNormal, 0x20: segCloseNormal, 0x30: segCloseNormal, 0x32: segCloseNormal, 0x34: segCloseNormal, 0x36: segCloseNormal, 0x40: segCloseNormal},
+		0x41: {0x10: segCloseNormal, 0x14: segCloseNormal, 0x17: segCloseNormal, 0x20: segCloseNormal, 0x30: segCloseNormal, 0x32: segCloseNormal, 0x34: segCloseNormal, 0x36: segCloseNormal, 0x40: segCloseEventID},
+		0x50: {0x10: segCloseNormal, 0x14: segCloseNormal, 0x17: segCloseNormal, 0x20: segCloseNormal, 0x30: segCloseNormal, 0x32: segCloseNormal, 0x34: segCloseNormal, 0x36: segCloseNormal, 0x40: segCloseUnconditional, 0x50: segCloseNormal},
+		0x51: {0x10: segCloseNormal, 0x14: segCloseNormal, 0x17: segCloseNormal, 0x20: segCloseNormal, 0x30: segCloseNormal, 0x32: segCloseNormal, 0x34: segCloseNormal, 0x36: segCloseNormal, 0x40: segCloseUnconditional, 0x50: segCloseEventID},
+	}
+	// add program breakaway rules to the close map.  Only ProgramResumption,
+	// Unscheduled Event and Network signals can exit breakaway.
+	// Program resumption should not close ProgramBreakaway as this logic will be
+	// handled in state.  We want processing to stop so that it won't close
+	// events after the breakaway
+	segCloseRules[SegDescUnscheduledEventStart][0x13] = segCloseNormal
+	segCloseRules[SegDescUnscheduledEventEnd][0x13] = segCloseNormal
+	segCloseRules[SegDescNetworkStart][0x13] = segCloseNormal
+	segCloseRules[SegDescNetworkEnd][0x13] = segCloseNormal
+}
+
+func (d *segmentationDescriptor) parseDescriptor(data []byte) error {
+	buf := bytes.NewBuffer(data)
+	// closure to ignore EOF error from buf.ReadByte().  We've already checked
+	// the length, we don't need to continually check it after every ReadByte
+	// call
+	readByte := func() byte {
+		b, _ := buf.ReadByte()
+		return b
+	}
+	if binary.BigEndian.Uint32(buf.Next(4)) != segDescID {
+		return mpegts.ErrSCTE35InvalidDescriptorID
+	}
+	d.eventID = binary.BigEndian.Uint32(buf.Next(4))
+	if readByte()&0x80 == 0 { // Cancel indicator
+		flags := readByte()
+		if flags&0x80 == 0 {
+			// skip over component info
+			ct := readByte()
+			if int(ct)*6 > buf.Len()-5 {
+				return mpegts.ErrInvalidSCTE35Length
+			}
+			for ; ct > 0; ct-- {
+				buf.Next(6)
+			}
+		}
+		d.hasDuration = flags&0x40 != 0
+		if d.hasDuration {
+			if buf.Len() < 10 {
+				return mpegts.ErrInvalidSCTE35Length
+			}
+			d.duration = uint40(buf.Next(5))
+		}
+		// upid unneeded now...
+		d.upidType = SegUPIDType(readByte())
+		upidLen := int(readByte())
+		if buf.Len() < upidLen+3 {
+			return mpegts.ErrInvalidSCTE35Length
+		}
+		d.upid = buf.Next(upidLen)
+		d.typeID = SegDescType(readByte())
+		d.segNum = readByte()
+		d.segsExpected = readByte()
+	}
+	return nil
+}
+
```

Comment:
IIRC GoDoc will format for any exported functions/fields in a package.  Take a look at the GoDoc pages for the rest of the package and check out how they format.  Let's just get things so it follows suit for consistency.